internal/merkle/proof_test.go

```diff
@@ -247,3 +971,86 @@ func (s *ComputeProofSuite) TestItMatchesMachineImplementation() {
 		root,
 	)
 }
+
+func FuzzVerifyProofs(f *testing.F) {
+	f.Add(uint(10), uint(10), uint(5))
+	f.Fuzz(func(t *testing.T, height, leafCount, leafIdx uint) {
+		height = bound(height, 3, 20)
+		leafCount = bound(leafCount, 1, uint(math.Pow(2, float64(height))))
```

Comment:
I think we can use bit shifting here.

```suggestion
		leafCount = bound(leafCount, 1, 1 << height)
```

Reply:
🤯 🔫

Reply:
Done at 26708fc3d1d198807daf41d1303e070ca5048a90